modules/apps/portal/portal-log4j/src/main/java/com/liferay/portal/log4j/internal/Log4jConfigUtil.java

```diff
@@ -75,6 +76,13 @@ public static Map<String, String> configureLog4J(
 			for (Element element : rootElement.elements()) {
 				_removeAppender(element, removedAppenderNames);
 
+				if (_isEnabledCompanyWebIdLogContext(element)) {
```

Comment:
@yuhai Is it possible to avoid doing this, and bind the registration with the specific appender? Which is to say, as long as the appender is used, you should register this context. This way, you don't need to manually check the xml file. Just let Lo4J read the xml file and take care of appender creation, and you register the context in that appender creation with a one-time protection.

Reply:
@dantewang, yes, I will do that.